src/sourcerer/infrastructure/storage_provider/services/azure.py

```
@@ -0,0 +1,233 @@
+            container = storage_path_parts[0]
+
+            storage_path = storage_path_parts[1] if len(storage_path_parts) > 1 else ""
+            blob_name = os.path.join(storage_path, dest_path or source_path.name)
+
+            blob_client = containers_client.get_container_client(container)
+            with open(source_path, "rb") as file_handle:
+                blob_client.upload_blob(blob_name or source_path.name, file_handle, overwrite=True)
+        except Exception as ex:
+            raise UploadStorageItemsException(str(ex)) from ex
+
+    def download_storage_item(self, storage: str, key: str, progress_callback: Callable | None = None) -> str:
```

Ruff (0.8.2)

Unused method argument: `progress_callback`

(ARG002)